### PR TITLE
[train] New persistence mode: `Tune tests + examples (small)` (batch 1)

### PR DESCRIPTION
**Pull request by justinvyu**
## Why are these changes needed?

This PR updates a bunch of Tune tests -- mostly tune controller execution tests.

## Related issue number

https://github.com/ray-project/ray/issues/38570

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/tune/BUILD [hunk 1]
```diff
@@ -94,15 +94,6 @@ py_test(
     tags = ["team:ml", "exclusive", "rllib"],
 )
 
-py_test(
-    name = "test_cluster_searcher",
```
Comment (justinvyu) [inline python/ray/tune/BUILD:98]: This test is pretty much replaced by `test_experiment_restore`. The test setup was strange and hard to fix.

File: python/ray/train/tests/util.py [hunk 2]
```diff
@@ -22,3 +23,11 @@ def load_dict_checkpoint(checkpoint: Checkpoint) -> Dict[str, Any]:
     with checkpoint.as_directory() as checkpoint_dir:
         with open(os.path.join(checkpoint_dir, "data.pkl"), "rb") as f:
             return ray_pickle.load(f)
+
+
+def mock_storage_context() -> StorageContext:
+    return StorageContext(
+        storage_path=tempfile.mkdtemp(),
```
Comment (ericl) [inline python/ray/train/tests/util.py:30]: Will this get cleaned up eventually when the test finishes?

**Review by ericl — APPROVED** [review]

File: python/ray/train/tests/util.py (hunk 2 above)
Comment (justinvyu) [reply python/ray/train/tests/util.py:30]: No, we need this thing for tests involving `TuneController`

**Review by zhe-thoughts — APPROVED** [review]
Necessary Train change for 2.7

**Pull request merged**